### ISSUE TITLE
Use of `unicode` and `basestring`

### ISSUE THREAD

**Issue by happz** [teemtee/tmt#1072]
These two names are used often in `tmt.utils`, e.g. here: https://github.com/teemtee/tmt/blob/main/tmt/utils.py#L617

IIUIC, these are used mostly thanks to from where the code comes, but actually, they have no effect at all: every time `isinstance(text, unicode)` attempts to run, raises an exception. Code is guarded against this kind of behavior, making both `basestring` and `unicode` presence in code pointless, complicating the reasoning about how code behaves. It's not a serious issue, but it makes the addition of type annotations more complicated than it needs to be.

So, unless there's a reason for keeping them around, I'd propose getting rid of `basestring` and `unicode`, simplifying the related code. I can submit a patch, seeking blessing of developers more familiar with the code - @psss, @lukaszachy, @thrix?

**Comment by thrix** [comment]
@happz makes sense to me, but I would wait for @psss, as I guess this part of code was added for a reason?

**Comment by happz** [comment]
> @happz makes sense to me, but I would wait for @psss, as I guess this part of code was added for a reason?

Definitely for a reason, but my impression was it's more of a copy & paste issue, to bootstrap functionality (it's a structured field implementation). TMT does not support running on top of Python 2, and these classes simply don't exist in Python 3, therefore this Python 2 code that did play important role in places the code came from isn't necessary for the TMT codebase.

No worries, I'm not touching anything, yet :)

**Comment by psss** [comment]
Yes, these are mostly leftovers from the `StructuredField` implementation. It should be safe to remove them as `tmt` is Python 3 only.
